dsa-23au/java-dsa/sorting/src/main/java/dev/codewithfriends/App.java

```diff
@@ -33,10 +33,13 @@ public static int[] mergeSort(int[] array){
 
     public static int[] mergeSortHelper(int[] arr, int start, int end){
         int[] newArray = new int[end - start + 1]; 
+        int length = end - start + 1;
+
+        mergeSortHelper(arr, start, start + (length / 2));       // left half
+        mergeSortHelper(arr, start + (length / 2), end + 1);     // right half
```

Comment:
Good start. The start and end indices for each call to `mergeSortHelper` are inclusive, so these two calls above are repeating the index `start + (length /2)` and overextending to the index `end` which may not exist.

Consider the second call being
`        mergeSortHelper(arr, start + (length / 2)+1, end);     // right half`